src/main/java/org/jitsi/videobridge/RtpChannel.java

```diff
@@ -1991,6 +1968,121 @@ public RtpChannelTransformEngine getTransformEngine()
         return this.transformEngine;
     }
 
+    /**
+     * Creates the {@code MediaStreamTrack}s from signaling and adds them to the
+     * {@code MediaStream} that is associated to this {@code RtpChannel}.
+     *
+     * @param sources  The <tt>List</tt> of <tt>SourcePacketExtension</tt> that
+     * describes the list of sources of this <tt>RtpChannel</tt> and that is
+     * used as the input in the update of the Sets the <tt>Set</tt> of the SSRCs
+     * that this <tt>RtpChannel</tt> has signaled.
+     * @param sourceGroups
+     */
+    public void setMediaStreamTracks(
+        List<SourcePacketExtension> sources,
+        List<SourceGroupPacketExtension> sourceGroups)
+    {
+        if ((sources == null || sources.isEmpty())
+            && (sourceGroups == null || sourceGroups.isEmpty()))
+        {
+            return;
+        }
+
+        this.setSources(sources); // TODO remove and rely on MSTs.
+        this.setSourceGroups(sourceGroups); // TODO remove and rely on MSTs.
+
+        Map<Long, MediaStreamTrack> tracksBySSRC = new TreeMap<>();
+        Map<Long, long[]> encodings = new TreeMap<>();
+
+        List<Long> freeSSRCs = new ArrayList<>();
+
+        if (sources != null && sources.size() != 0)
```

Comment:
And why do we have to write these checks twice. Cannot we write them once and use local variables?